src/utilities/tree_sitter_utilities.rs

```diff
@@ -303,6 +304,21 @@ pub(crate) fn get_replace_range(input_edit: InputEdit) -> Range {
   }
 }
 
+pub(crate) fn get_ts_query_parser() -> Parser {
+  let mut parser = Parser::new();
+  parser
+    .set_language(tree_sitter_query::language())
+    .expect("Could not set the language for the parser.");
```

Comment:
I don't understand what's going on here. Are we creating a new parser and expecting it to be set to an invalid language? Why? (This at the very least needs documentation)

Reply:
aah maybe its just bad api naming in rustlang. 
We are setting a language, and if the setter throws an exception we panic with "_Could not set the language for the parser._"

.expect() ~= .orElseThrow() in Java

Reply:
Got it. Also, a bit surprised that `tree_sitter_query` doesn't need any import. For a second I thought this was of the same type `piranha_args.language()`, i.e. `PiranhaLanguage`, and was wondering why this wasn't simply `tree_sitter_query::language().parser()`, but I see the difference now. I still think this all merits a short comment here explaining what this code does (uses tree sitter to construct a parser for its own DSL without all the extensions we have made to Piranha-supported language, correct?)

Reply:
aah i see the the source of confusion.  
Yes u r correct, we are using ts to parse its own dsl.
I will address this as a comment in the code